Improve property access and added post processing

andreasschacht opened the pull request:
Hi, 
I made some improvements, e.g. access to private/protected properties. I got a request to make a PR ans ask you, if you can merge and create a new release. So others can also use this approvements and install them via packagist. 

Best regards

Spea commented on composer.json:
```diff
@@ -17,18 +17,18 @@
         "symfony/options-resolver": "~2.1",
-        "symfony/filesystem": "~2.1",
-        "phpcollection/phpcollection": "0.2.*",
+        "symfony/filesystem": "3.0.*",
```
Please use `"~2.1 | ~3.0"`

Spea commented on composer.json:
```diff
@@ -17,18 +17,18 @@
+        "phpcollection/phpcollection": "~0.4",
         "symfony/property-access": "~2.2",
-        "jms/cg": "1.0.0"
+        "jms/cg": ">=1.0.0"
```
This would also allow version 2, 3, etc. So you should use `~1.0` instead

Spea commented on src/Sp/FixtureDumper/Generator/AbstractGenerator.php:
```diff
@@ -50,6 +50,23 @@
+    protected $skipNullValues = false;
+
+
+
```
Remove two extra lines

Spea commented on src/Sp/FixtureDumper/Generator/FieldNamesPostProcessorInterface.php:
```diff
@@ -0,0 +1,46 @@
+ * User: andreasschacht
+ * Date: 01.08.16
+ * Time: 11:27
+ */
```
Remove this comment block

Spea commented on src/Sp/FixtureDumper/Generator/FieldNamesPostProcessorInterface.php:
```diff
@@ -0,0 +1,46 @@
+     * @return array
+     */
+    public function getData($object, ClassMetaData $metadata, array $data);
+}
```
Please make sure that you always add a new line at the end of a file. Since you use PHPStorm, there is an option you can activate

![image](https://cloud.githubusercontent.com/assets/495017/21263394/ce3339b4-c396-11e6-9de3-f3cc99b8a6ba.png)

Spea commented on src/Sp/FixtureDumper/Generator/PropertyReaderInterface.php:
```diff
@@ -0,0 +1,33 @@
+     * @return mixed
+     */
+    public function getValue($object, $property);
+}
```
Same here

Spea commented on src/Sp/FixtureDumper/Generator/FieldNamesPostProcessorInterface.php:
```diff
@@ -0,0 +1,46 @@
+     *
+     * @return boolean true, if this class can process the data
+     */
+    public function isSupporting($object, ClassMetaData $metadata, array $data);
```
Would call this method just `supports`

Spea commented:
Thx for this PR, but make sure that the tests are running again ;)

andreasschacht commented:
Thanks, I fixed it and renamed the method name. I had to restrict the jms/cg version to be less than v1.2, because this version breaks your tests. It adds a newline to the DocBlock of the generated class (with the ClassFixtureGenerator), the versions before don't. So your string comparison will fail.

Spea commented:
https://github.com/Spea/fixture-dumper/releases/tag/v1.1.0 Here you go :)
